packages/x-data-grid/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -456,6 +456,10 @@ export const useGridColumnResize = (
 
   const storeReferences = (colDef: GridStateColDef, separator: HTMLElement, xStart: number) => {
     const root = apiRef.current.rootElementRef.current!;
+    // HACK: The jsdom test is failing because CSS.escape doesn't exist in that context.
+    // We have decided that we want to preserve the bug in jsdom
+    // https://github.com/mui/mui-x/pull/13069#discussion_r1603270444
+    const escapedColDefField = typeof CSS === 'undefined' ? colDef.field : CSS.escape(colDef.field);
```

Comment:
This is inconsistent with #2033. To normalize to have one solution for one problem.

Reply:
Agreed, use the existing `escapeOperandAttributeSelector` function instead of `CSS.escape`.

Reply:
Why not standardize on CSS.escape?

Reply:
Not available in jsdom.

Reply:
Alright, I have added this commit ed3472b0e2b38058bc8964edfd9d8353c392a4c5 as the behavior might not be 100% correct right now looking at https://github.com/mathiasbynens/CSS.escape/blob/master/css.escape.js.

Reply:
I did argue we should include a ponyfill for `CSS.escape` but I was defeated. Read the full discussion [here](https://mui-org.slack.com/archives/C011VC970AW/p1715772065813379).

Reply:
Right, so from what I understand, there are two distinct discussions:

1. Using `escapeOperandAttributeSelector` is wrong, we should use `CSS.escape`. But `escapeOperandAttributeSelector` has been good enough so far, so we continue with is, waiting for the problem to be surfaced
2. jsdom doesn't support `CSS.escape`, should the data grid do something about it? The strategy used so far has been to support jsdom with no need for custom setup for developers. e.g. #12983.